ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureRequest.java

```diff
@@ -167,10 +167,14 @@ public void setExemptionRequested(boolean exemptionRequested) {
      * Optional. An authentication created using this flag should only be used for adding a
      * payment method to the merchant's vault and not for creating transactions.
      *
-     * @param cardAddChallengeRequested If set to true, the authentication challenge will be requested from the issuer
-     *                to confirm adding new card to the merchant's vault.
+     * @param cardAddChallengeRequested If set to true, the authentication challenge will be requested
+     *                                  from the issuer to confirm adding new card to the merchant's
+     *                                  vault. If not set and amount is 0, the authentication challenge
+     *                                  will be presented to the user. If set to false, when the amount
+     *                                  is 0, the authentication challenge will not be presented to the user.
+     *
      */
-    public void setCardAddChallengeRequested(boolean cardAddChallengeRequested) {
+    public void setCardAddChallengeRequested(@Nullable Boolean cardAddChallengeRequested) {
```

Comment:
I can't remember exactly with the whole Kotlin conversion, but it'd be nice if we could still use `boolean` in the merchant facing API and store the variable in a `Boolean` internally.

Reply:
My thought was the getter has to be able to return null because the param can be null and doesn't the setter need to align with the getter for synthesized properties. Is there another way around that?

Reply:
Yeah you're right I agree. The possibility of a collision with Kotlin property synthesis makes this our best option.

Reply:
`If not set and amount is 0, the authentication challenge will be presented to the user.`

Isn't this a breaking change?

Reply:
It sounds like default behavior to me but I'm not sure. I wonder if this still holds true with the current version since this property not being set is equivalent to the property not existing.

Reply:
Thinking again about this, it's pretty confusing to me. It seems like a limitation of the GW API - I take this as if we pass `"card_add"` to the GW without a bool value and the amount is 0, it automatically induces a challenge? I would consider this a breaking change.

We could either talk to the GW team and get them to fix this before we put this in the SDKs .... OR .... simply not include the `"card_add"` param in the request to the GW unless the merchant has set it as true or false. This is assuming my understanding of how this is working, is correct! Let me know!

Reply:
@sshropshire - agreed, the behavior of this property not being set should be equivalent to it not existing in previous SDK versions.

Reply:
Yeah tbh I'm not 100% clear on the gateway behavior either. On our side we do make `null` the default value by using a boxed type `Boolean` instead of a primitive. Then we do a null check before adding the `card_add` parameter to the payload.

Reply:
Ok we are going to follow up with 3DS to confirm the GW behavior on this and make sure it's not a breaking change. Let's hold off on merging this / releasing the previous changes until we get that confirmation.

Reply:
> Ok we are going to follow up with 3DS to confirm the GW behavior on this and make sure it's not a breaking change. Let's hold off on merging this / releasing the previous changes until we get that confirmation.

Thanks for looking into this. Is this something that can be tested manually (by setting a 0 amount) in the SDK itself and going through a 3DS flow?

Reply:
Not sure about manual testing - when I set amount to 0 I get a nonce but can't create a transaction (as expected because this behavior is for creating payment method not transaction). 

But looks like we have confirmation that this GW behavior was introduced over a year ago so we are just adding the true/false option to give merchants more control over existing behavior.

Reply:
Rad - thanks for confirming this, Sarah!